src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -104,321 +150,710 @@ enum Spells
+
+            if (id == POINT_PREFIGHT)
             {
-                if (uiRenewTimer <= uiDiff)
+                me->SetWalk(false);
+                me->GetMotionMaster()->MoveTargetedHome(); // this corrects facing 100%
+                if (me->GetEntry() == NPC_ARGENT_MONK)
                 {
-                    me->InterruptNonMeleeSpells(true);
-                    uint8 uiTarget = urand(0, 1);
-                    switch (uiTarget)
+                    if (Creature* lightWielder = me->FindNearestCreature(NPC_ARGENT_LIGHWIELDER, 15.0f))
                     {
-                        case 0:
-                            DoCast(me, SPELL_RENEW);
-                            break;
-                        case 1:
-                            if (Creature* pMemory = ObjectAccessor::GetCreature(*me, MemoryGUID))
-                                if (pMemory->IsAlive())
-                                    DoCast(pMemory, SPELL_RENEW);
-                            break;
+                        if (lightWielder->GetFollowAngle() != 0.f)
+                        {
+                            lightWielder->GetMotionMaster()->Clear();
+                            lightWielder->GetMotionMaster()->MovePoint(0, lightWielder->GetHomePosition());
+                        }
                     }
-                    uiRenewTimer = urand(15000, 17000);
-                } else uiRenewTimer -= uiDiff;
+
+                    if (Creature* priestess = me->FindNearestCreature(NPC_PRIESTESS, 15.0f))
+                    {
+                        if (priestess->GetFollowAngle() != 0.f)
+                        {
+                            priestess->GetMotionMaster()->Clear();
+                            priestess->GetMotionMaster()->MovePoint(0, priestess->GetHomePosition());
+                        }
+                    }
+                }
             }
+        }
 
-            if (!bHealth && !HealthAbovePct(25))
+        void JustSummoned(Creature* summon) override
+        {
+            if (me->GetEntry() == NPC_PRIESTESS)
+                _fountainGuid = summon->GetGUID();
+        }
+
+        void DamageTaken(Unit* /*attacker*/, uint32& damage) override
+        {
+            if (me->GetEntry() == NPC_ARGENT_MONK && damage >= me->GetHealth() && !_shielded)
             {
-                me->InterruptNonMeleeSpells(true);
-                DoCastAOE(SPELL_HOLY_NOVA, false);
-                DoCast(me, SPELL_SHIELD);
-                DoCastAOE(SPELL_SUMMON_MEMORY, false);
-                DoCastAOE(SPELL_CONFESS, false);
+                _shielded = true;
+                damage = me->GetHealth() - 1;
+                me->AttackStop();
+
+                DoCast(SPELL_DIVINE_SHIELD);
+                DoCast(SPELL_FINAL_MEDITATION);
+            }
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            _instance->SetData(DATA_ARGENT_SOLDIER_DEFEATED, _instance->GetData(DATA_ARGENT_SOLDIER_DEFEATED) + 1);
```

Comment:
Is there any reason to invoke `GetData` here? Is there ever a situation where `DATA_ARGENT_SOLDIER_DEFEATED` is modified, except increments by 1?